RecoMTD/TrackExtender/plugins/TrackExtenderWithMTD.cc

```diff
@@ -112,23 +108,23 @@ namespace {
       return nSegment_;
     }
 
-    inline const double computeTof(double mass_inv2) {
-      double tof(0.);
-      for (size_t iSeg = 0; iSeg < nSegment_; iSeg++) {
-        double gammasq = 1. + segmentMom2_[iSeg] * mass_inv2;
-        double beta = std::sqrt(1. - 1. / gammasq);
+    inline double computeTof(double mass_inv2) const {
```

Comment:
uhm, the suggestion to use float implied that the return is also going to be changed to a float

Reply:
unless the calling code is modified, in any case the float will be implicitly converted into a double there. In our code guidelines we advice to use C++ type casting (https://cms-sw.github.io/cms_coding_rules.html#4--technical-coding-rules-1 point 15), and in general I understand that explicit casting is favoured wrt implicit one. Your suggestion can be trivially done, but I'd like to understand what is the net benefit, unless a larger part of the code is converted into ```float``` precision.

Reply:
here I made a comment in the context of a new function, without looking at the larger context that uses double.
Admittedly, having a mix of double and float without care is a worse situation than making double precision computations in places where floats are good enough.